src/platform/packages/shared/kbn-esql-ast/src/commands_registry/commands/eval/validate.test.ts

```diff
@@ -378,13 +362,22 @@ describe('EVAL Validation', () => {
       evalExpectErrors(`from a_index | eval col0 = dateField - 1 ${unit.toUpperCase()}`, []);
       evalExpectErrors(`from a_index | eval col0 = dateField - 1 ${capitalize(unit)}`, []);
       evalExpectErrors(`from a_index | eval col0 = dateField + 1 ${unit}`, []);
-      evalExpectErrors(`from a_index | eval 1 ${unit} + 1 year`, []);
```

Comment:
This is cool... with the old logic, we were incorrectly allowing this case. But now that we've tightened up the typing, the validator correctly flags it based on the `+` function definition